src/payment/strategies/square/square-payment-strategy.ts

```diff
@@ -106,24 +131,128 @@ export default class SquarePaymentStrategy extends PaymentStrategy {
+                    }
+
+                    return {
+                        requestShippingAddress: true,
+                        requestBillingInfo: true,
+                        currencyCode: storeConfig.currency.code,
+                        countryCode: 'US',
+                        total: {
+                            label: storeConfig.storeProfile.storeName,
+                            amount: checkout.subtotal.toString(),
+                            pending: false,
+                        },
+                    };
+                },
+
+                validateShippingContact: (errors: SquareValidationErrors) => {
+                    if (errors) {
+                        this._handleSquareValidationErrors(errors);
+                    }
                 },
             },
         };
     }
 
-    private _cardNonceResponseReceived(errors: any, nonce: string): void {
+    private _paymentInstrumentSelected(methodId: string) {
+        return this._store.dispatch(this._paymentStrategyActionCreator.widgetInteraction(() => {
+                return Promise.all([
+                    this._store.dispatch(this._checkoutActionCreator.loadCurrentCheckout()),
+                    this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(methodId)),
+                ]);
+        }, { methodId }), { queueId: 'widgetInteraction' });
+    }
+
+    private _handleNonceGenerationErrors(errors: NonceGenerationError[]) {
+        const error = errors[0];
+
+        throw new StandardError(error.message);
+    }
+
+    private _handleSquareValidationErrors(error: SquareValidationErrors) {
+        if (!error.country && !error.region && !error.city && !error.addressLines && !error.postalCode) {
+            throw new StandardError('Unknown error');
+        }
+
+        if (error.country) {
+            throw new StandardError(error.country.join(','));
+        }
+        if (error.region) {
+            throw new StandardError(error.region.join(','));
+        }
+        if (error.city) {
+            throw new StandardError(error.city.join(','));
+        }
+        if (error.addressLines) {
+            throw new StandardError(error.addressLines.join(','));
+        }
+
+        throw new StandardError(error.postalCode.join(','));
+    }
+
+    private _cardNonceResponseReceived(nonce: string, nonceGenerationErrors: NonceGenerationError[] | undefined): void {
         if (!this._deferredRequestNonce) {
             throw new StandardError();
         }
 
-        if (errors) {
-            this._deferredRequestNonce.reject(errors);
+        if (nonceGenerationErrors) {
+            this._deferredRequestNonce.reject(nonceGenerationErrors);
         } else {
             this._deferredRequestNonce.resolve({ nonce });
         }
     }
+
+    private _setExternalCheckoutData(cardData: CardData, nonce: string): Promise<Response> {
+        const url = `checkout.php?provider=squarev2&action=set_external_checkout`;
```

Comment:
I see we do this for ChasePay as well, but should we be hard-coding these URLs in the library? Possibly a @bigcommerce/payments question.